server/src/main/java/org/apache/druid/client/selector/ServerSelector.java

```diff
@@ -162,12 +163,20 @@ public boolean isEmpty()
   @Nullable
   @Override
   public QueryableDruidServer pick()
+  {
+    if (!historicalServers.isEmpty()) {
+      return strategy.pick(historicalServers, segment.get());
+    }
+    return strategy.pick(realtimeServers, segment.get());
+  }
+
+  public <T> QueryableDruidServer pick(Query<T> query)
```

Comment:
It seems that `pick()` is no longer in use. I think we can remove `pick()` but keep only this new one if `ServerSelector` doesn't implement `DiscoverySelector`.

Reply:
I removed it, the method was called a couple of times in tests (which seems OK) and in the etag creation part, which seems OK to me. It looks a little funny to pass in null as a single parameter, but I could argue either way vs having a method that isn't used elsewhere.